sage.py

```diff
@@ -481,1274 +423,920 @@ def getepisodes(action_seq, mcat, plot, debug=False):
-                    if attack.split("|")[:2] == vname.split("|")[:2]: # if it's the objective
-                        if len(sub_attempt) <= 1: ## If only a single node, reject
-                            sub_attempt = []
-                            continue
-                        attempts.append(sub_attempt)
-                        sub_attempt = []
-                        observed_obj.add(vname)
-                        continue
-                team_attacker = att.split('->')[0] # team+attacker
-                if team_attacker not in team_level.keys():
-                    team_level[team_attacker] = []
-                    
-                team_level[team_attacker].extend(attempts)
-                #team_level[team_attacker] = sorted(team_level[team_attacker], key=lambda item: item[1])
-            #print(observed_obj)
-            # print('elements in graph', team_level.keys(), sum([len(x) for x in team_level.values()]))
-
-            if sum([len(x) for x in team_level.values()]) == 0: # if no team obtains this objective or targets this victim, don't generate its AG.
+    objectives = _get_objective_nodes(state_sequences)
+
+    for victim in total_victims:
+        print('!!! Rendering AGs for Victim', victim)
+        for objective in objectives:
+            # Get the variants of current objective and attempts per team
+            team_attack_attempts, observed_obj = _get_attack_attempts(state_sequences, victim, objective, in_main_model)
+
+            # If no team has obtained this objective or has targeted this victim, don't generate its AG
+            if sum([len(attempt) for attempt in team_attack_attempts.values()]) == 0:
                 continue
 
-            AGname = attack.replace('|', '').replace('_','').replace('-','').replace('(','').replace(')', '')
-            lines = []
-            lines.append((0,'digraph '+ AGname + ' {'))
-            lines.append((0,'rankdir="BT"; \n graph [ nodesep="0.1", ranksep="0.02"] \n node [ fontname=Arial, fontsize=24,penwidth=3]; \n edge [ fontname=Arial, fontsize=20,penwidth=5 ];'))
-            root_node = translate(attack, root=int_victim)
-            lines.append((0, '"'+root_node+'" [shape=doubleoctagon, style=filled, fillcolor=salmon];'))
-            lines.append((0, '{ rank = max; "'+root_node+'"}'))
-            
-            for obj in list(observed_obj): # for each variant of objective, add a link to the root node, and determine if it's sink
-                lines.append((0,'"'+translate(obj)+'" -> "'+root_node+'"'))
-                
-                sinkflag = False
-                for sink in sev_sinks:    
-                    if obj.split("|")[-1] == sink:
-                        sinkflag = True
-                        break
-                if sinkflag:
-                    lines.append((0,'"'+translate(obj)+'" [style="filled,dotted", fillcolor= salmon]'))
+            print('\t!!!! Objective', objective)
```

Comment:
I have moved the print for Objective lower, so that it is only printed if the AG is actually created. This makes it easier to see from the logs for each victim which AGs have been created and for which victim no AGs have been created

Reply:
Maybe keep this line as it is, and once the graph is actually created, we print "created". Then, at least we know the code is checking these objective options even if it doesnt save them.

Reply:
Something like this, or do you also want to print if it is *not* created?
![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/66218bef-2e8e-4aeb-b24f-ccc8c168d55d)

Reply:
[Updated as discussed in the meeting]